### PR TITLE
Various bugs

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: GenericTrigger.lua [hunk 1]
```diff
@@ -855,13 +855,27 @@ do
...
+  ["Combo Points"] = 4,
+  ["Holy Power"] = 9,
+  ["Insanity"] = 13,
+  ["Chi Power"] = 12,
+  ["Astral Power"] = 8,
+  ["Maelstrom"] =  11,
+  ["Arcane Charges"] = 16,
+  ["Fury"] = 17,
+  ["Pain"] = 18,
+  ["Shards"] = 7,
```
Comment (Stanzilla) [inline GenericTrigger.lua]: *Soul

File: GenericTrigger.lua [hunk 2]
```diff
@@ -855,13 +855,27 @@ do
...
   ["source"] = "sourceName",
   ["destunit"] = "destUnit",
   ["dest"] = "destName"
 }
 
+local oldPowerTriggers = {
+  ["Combo Points"] = 4,
+  ["Holy Power"] = 9,
+  ["Insanity"] = 13,
+  ["Chi Power"] = 12,
```
Comment (Stanzilla) [inline GenericTrigger.lua]: -Power

File: Types.lua [hunk 3]
```diff
@@ -343,15 +343,25 @@ WeakAuras.subevent_suffix_types = {
...
-  [13] = L["Insanity"]
+  [7] = L["Soul Shards"],
+  [8] = L["Astral Power"],
+  [9] = L["Holy Power"],
+  [11] = L["Maelstrom"],
+  [12] = L["Chi"],
+  [13] = L["Insanity"],
+  [16] = L["Arcane Charges"],
+  [17] = L["Fury"],
+  [18] = L["Pain"],
```
Comment (Stanzilla) [inline Types.lua]: I think we can/should use the constants for translation, like INSANITY etc

File: Prototypes.lua [hunk 4]
```diff
@@ -17,9 +17,6 @@ function WeakAuras.IsSpellInRange(spellId, unit)
   return SpellRange.IsSpellInRange(spellId, unit);
 end
 
--- GLOBALS: SPELL_POWER_CHI SPELL_POWER_INSANITY SPELL_POWER_LUNAR_POWER SPELL_POWER_MAELSTROM SPELL_POWER_ARCANE_CHARGES
--- GLOBALS: SPELL_POWER_FURY SPELL_POWER_PAIN SPELL_POWER_MANA
```
Comment (Stanzilla) [inline Prototypes.lua]: add the constants here, too, for the linter

File: BuffTrigger.lua [hunk 5]
```diff
@@ -1052,12 +1058,20 @@ do
...
+    elseif(event == "NAME_PLATE_UNIT_ADDED") then
+      uidTrack(...);
+    elseif(event == "NAME_PLATE_UNIT_REMOVED") then
+      local unit = ...
+      WeakAuras.ReleaseUID(unit);
+      unit = unit.."target";
+      WeakAuras.ReleaseUID(unit);
     elseif(event == "UNIT_AURA") then
+      -- Note: Now using UNIT_AURA in addition to COMBAT_LOG_EVENT_UNFILTERED
+      --  * UNIT_AURA because there is no combat log event when an aura gets refrshed by a spell (bug?).
```
Comment (Stanzilla) [inline BuffTrigger.lua]: Tried _AURA_APPLIED_DOSE or _AURA_REFRESH?

Comment (InfusOnWoW) [inline BuffTrigger.lua]: Ah teh comment is old, I just moved it

File: BuffTrigger.lua [hunk 6]
```diff
@@ -963,13 +962,14 @@ do
               -- Shouldn't affect duration/expirationTime nor icon
               data.GUIDs[destGUID].duration = data.GUIDs[destGUID].duration or 0;
               data.GUIDs[destGUID].expirationTime = data.GUIDs[destGUID].expirationTime or math.huge;
-              data.GUIDs[destGUID].icon = data.GUIDs[destGUID].icon or nil;
+              data.GUIDs[destGUID].icon = data.GUIDs[destGUID].icon or (spellId and select(3, GetSpellInfo(spellId)));
```
Comment (Stanzilla) [inline BuffTrigger.lua]: can we avoid the 2 selects and local the value before the if statements?

File: Types.lua [hunk 7]
```diff
@@ -4,6 +4,12 @@ local L = WeakAuras.L;
 local LSM = LibStub("LibSharedMedia-3.0");
 local LBR = LibStub("LibBabble-Race-3.0"):GetLookupTable()
 
+-- GLOBALS:   MANA RAGE FOCUS ENERGY COMBO_POINTS RUNIC_POWER SOUL_SHARDS LUNAR_POWER HOLY_POWER MAELSTROM CHI INSANITY ARCANE_CHARGES FURY PAIN
+
+local wipe, tinsert = wipe, tinsert
+local  GetNumShapeshiftForms, GetShapeshiftFormInfo = GetNumShapeshiftForms, GetShapeshiftFormInfo
```
Comment (Stanzilla) [inline Types.lua]: one space too much?

File: Types.lua [hunk 8]
```diff
@@ -4,6 +4,12 @@ local L = WeakAuras.L;
 local LSM = LibStub("LibSharedMedia-3.0");
 local LBR = LibStub("LibBabble-Race-3.0"):GetLookupTable()
 
+-- GLOBALS:   MANA RAGE FOCUS ENERGY COMBO_POINTS RUNIC_POWER SOUL_SHARDS LUNAR_POWER HOLY_POWER MAELSTROM CHI INSANITY ARCANE_CHARGES FURY PAIN
```
Comment (Stanzilla) [inline Types.lua]: one space too much?

**Pull request merged**